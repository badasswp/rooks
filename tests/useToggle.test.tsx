import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent, getByRole, getByTestId, render } from '@testing-library/react';

import Toggle from '../hooks/useToggle/Toggle';

describe('<UseToggle />', () => {
  it('should render the component with no issues', () => {
    const { container } = render(<Toggle isChecked={false} />);

    expect(container).toMatchSnapshot();
  });

  it('should toggle the state using the useToggle hook', () => {
    const { container } = render(<Toggle isChecked={true} />);

    fireEvent.click(getByRole(container, 'checkbox'));
    expect(getByTestId(container, 'status').textContent).toBe('Checked: false');

    fireEvent.click(getByRole(container, 'checkbox'));
    expect(getByTestId(container, 'status').textContent).toBe('Checked: true');
  });
});
