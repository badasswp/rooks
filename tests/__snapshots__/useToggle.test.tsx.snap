// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UseToggle /> should render the component with no issues 1`] = `
<div>
  <div>
    <p
      data-testid="status"
    >
      Checked: 
      false
    </p>
    <input
      type="checkbox"
    />
  </div>
</div>
`;
